core/src/main/java/org/infinispan/util/CoreBlockHoundIntegration.java

```diff
@@ -0,0 +1,113 @@
+      builder.allowBlockingCallsInside(RecoveryManagerImpl.class.getName(), "registerInDoubtTransaction");
+   }
+
+   private static void registerBlockingMethods(BlockHound.Builder builder) {
+      builder.markAsBlocking(CacheImpl.class, "size", "()I");
+      builder.markAsBlocking(CacheImpl.class, "size", "(J)I");
+      builder.markAsBlocking(CacheImpl.class, "containsKey", "(Ljava/lang/Object;)Z");
+      builder.markAsBlocking(CacheImpl.class, "get", "(Ljava/lang/Object;)Ljava/lang/Object;");
+      builder.markAsBlocking(CacheImpl.class, "put", "(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;");
+
+      // Distributed streams are blocking!
+      builder.markAsBlocking("org.infinispan.interceptors.distribution.DistributionBulkInterceptor$BackingEntrySet", "stream", "()Lorg/infinispan/CacheStream;");
+      builder.markAsBlocking("org.infinispan.interceptors.distribution.DistributionBulkInterceptor$BackingEntrySet", "parallelStream", "()Lorg/infinispan/CacheStream;");
+   }
+
+   private static void jgroupsMethodsAllowedToBlock(BlockHound.Builder builder) {
+      // UDP calls PlainDatagramSocketImpl#send which can block if OS buffer is full - but we just ignore
+      builder.allowBlockingCallsInside(UDP.class.getName(), "_send");
+      builder.allowBlockingCallsInside(FlowControl.class.getName(), "adjustCredit");
+      builder.allowBlockingCallsInside(Table.class.getName(), "add");
+      builder.allowBlockingCallsInside(Table.class.getName(), "get");
+
+      // Force Version class to be loaded before installing block hound (eagerly runs class initializer which blocks)
+      System.out.println("BlockHound in use with JGroups: " + Version.printVersion());
+
+      // The NonBlockingCreditMap being non blocking is a bit questionable as they hold the lock while doing a lot of
+      // work - including acquiring other locks and possibly calling await
+      builder.allowBlockingCallsInside(NonBlockingCreditMap.class.getName(), "decrement");
+      builder.allowBlockingCallsInside(NonBlockingCreditMap.class.getName(), "replenish");
+   }
+
+   private static void questionableMethodsAllowedToBlock(BlockHound.Builder builder) {
+      // Component registry has a lock to protect its state - is short lived lock
+      builder.allowBlockingCallsInside(BasicComponentRegistryImpl.class.getName(), "prepareWrapperChange");
+
+      // This one should probably not be allowed - it is waiting for another component to start
+      // TODO: This might actually be a bug in the rewiring logic. (shows in StateTransferOverwritingValueTest)
+      builder.allowBlockingCallsInside(BasicComponentRegistryImpl.class.getName(), "awaitWrapperState");
+
+      // Believe these can technically block for TCP when flushing to socket
+      builder.allowBlockingCallsInside(JGroupsTransport.class.getName(), "sendCommand");
+      builder.allowBlockingCallsInside(JGroupsTransport.class.getName(), "sendResponse");
```

Comment:
Do we need to add all the individual JGroups protocol methods as well, if we have these 2, or even just `JChannel#send`?